### PR TITLE
test: Add more unit tests for `BoundedVec` and `CyclicBoundedVec`

### PR DESCRIPTION
**Pull request by vadorovsky**
Some methods were not covered. This change achieves full coverage.

### CODE REVIEW

File: merkle-tree/bounded-vec/src/lib.rs [hunk 1]
```diff
@@ -809,6 +810,24 @@ mod test {
         bounded_vec
     }
 
+    #[test]
+    fn test_bounded_vec_metadata() {
```
Comment (ananas-block) [inline merkle-tree/bounded-vec/src/lib.rs:814]: This test tests metadata parsing/serialization right?
(It would be easier if the title reflects this.)

File: merkle-tree/bounded-vec/src/lib.rs [hunk 2]
```diff
@@ -819,6 +838,201 @@ mod test {
...
+        for i in 0..1000 {
+            let element = vec.get_mut(i).unwrap();
+            assert_eq!(element, &i);
+            *element = i * 2;
+        }
+        for i in 0..1000 {
+            assert_eq!(vec.get_mut(i), Some(&mut (i * 2)));
+        }
+        for i in 1000..10_000 {
+            assert!(vec.get_mut(i).is_none());
```
Comment (ananas-block) [inline merkle-tree/bounded-vec/src/lib.rs:925]: Just saw that capacity is just 1000, this tests only values outside of capacity to return None. Should we also test that values within capacity can be None?

File: merkle-tree/bounded-vec/src/lib.rs [hunk 3]
```diff
@@ -819,6 +838,201 @@ mod test {
...
+            assert!(vec.get(i).is_none());
+        }
+    }
+
+    #[test]
+    fn test_bounded_vec_get_mut() {
+        let mut vec = BoundedVec::with_capacity(1000);
+
+        for i in 0..1000 {
+            vec.push(i).unwrap();
```
Comment (ananas-block) [inline merkle-tree/bounded-vec/src/lib.rs:913]: We could add assert is none above this line.

File: merkle-tree/bounded-vec/src/lib.rs (hunk 2 above)
Comment (vadorovsky) [reply merkle-tree/bounded-vec/src/lib.rs:925]: Good idea, will do

**Review by ananas-block — APPROVED** [review]
